### DISCUSSION TITLE
how to reload slint listview

### DISCUSSION THREAD

**Discussion by jerrygou007** [slint-ui/slint#4107] [General]
in iOS，tableview.reload to refresh the list.

i can't find a way to reload the slint listview in slint side or rust side.

anyone know it?

**Comment by ogoffart** [comment]
Not exactly sure what you are trying to achieve, but the way to do it would be typically to set the model to a new model.
Or get the reset function of the model called.

**Reply by bjorn** [reply]
If your `hub-infos` are backed by a `VecModel`, then you can use [its `remove` function](https://slint.dev/releases/1.3.2/docs/rust/slint/struct.VecModel#method.remove) to remove an item from the model. It will then automatically be removed from the list view.

To reset the entire model, you can either create a new model instance and assign it to your `DataHub.hub-infos` global, or replace the contents of the existing model using [`set_vec`](https://slint.dev/releases/1.3.2/docs/rust/slint/struct.VecModel#method.set_vec). I'm doing this in my application as follows:

https://github.com/bjorn/raccoin/blob/304215415852eccae928259317227cb22c65d08e/src/main.rs#L1337-L1339

Though it's a bit more complicated than necessary, since I could not keep a `ModelRc` reference around in my app data due to some async code. I'd recommend keeping an `Rc<VecModel<...>>` reference around if possible.

(hmm, the post I've answered appears to be gone)
